### PR TITLE
Add user guide structure

### LINKED ISSUE #102: Update user guide structure

### PR DESCRIPTION
**Pull request by SamsonChew**
closes #102

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -4,16 +4,128 @@ title: "User Guide"
...
-<page-nav-print />
+Welcome to the **TrackMate User Guide** - your essential supplementary tools in your teaching journey as a SOC
+Teaching Assistant at National University of Singapore (NUS).
+
+In this comprehensive user guide, we will take you to experience a full journey with TrackMate step by step.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Table of Content
+
```
Comment (waihongteh) [inline docs/UserGuide.md:19]: Great enhancement to navigability through table of content, good job!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -4,16 +4,128 @@ title: "User Guide"
...
-<!-- * Table of Contents -->
-<page-nav-print />
+Welcome to the **TrackMate User Guide** - your essential supplementary tools in your teaching journey as a SOC
+Teaching Assistant at National University of Singapore (NUS).
+
+In this comprehensive user guide, we will take you to experience a full journey with TrackMate step by step.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Table of Content
```
Comment (Fui03) [inline docs/UserGuide.md:18]: I like how you organize and providing an overview of content!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -4,16 +4,128 @@ title: "User Guide"
...
+- [4.6 Saving the Data](#saving-the-data)
+- [4.7 Editing the Data File](#editing-the-data-file)
+
+[5. FAQ](#faq)
+
+[6. Known issues](#known-issues)
+
+[7. Glossary](#glossary)
+
+[8. Command summary](#command-summary)
```
Comment (Fui03) [inline docs/UserGuide.md:75]: Is it better if the command summary place it before [4. Commands] ?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -36,213 +148,261 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
...
-  e.g. if the command specifies `help 123`, it will be interpreted as `help`.
+In TrackMate, a parameter prefix acts as a delimiter for specifying different types of parameters in commands. 
+Here's a reference table for common parameter prefixes and their corresponding parameters:
 
-* If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
-  </box>
+TO BE DONE
 
-### Viewing help : `help`
+#### 3.3.2 Parameters:
```
Comment (Fui03) [inline docs/UserGuide.md:178]: I like how you've clearly presented the parameters! It makes it easy to understand and reference them!

File: docs/UserGuide.md [hunk 5]
```diff
@@ -4,16 +4,128 @@ title: "User Guide"
...
+- [3.1 Installation](#installation)
+- [3.2 Graphical User Interface Layout](#graphical-user-interface-layout)
+    - [3.2.1 User Interface Overview](#user-interface-overview)
+    - [3.2.2 Additional UI Components](#additional-ui-components)
+- [3.3 How to use TrackMate commands](#how-to-use-track-mate-nus-commands)
+    - [3.3.1 Parameter Prefixes](#parameter-prefixes)
+    - [3.3.2 Parameters](#parameters)
+    - [3.3.3 Command Format](#command-format)
+
+[4. Commands](#commands)
```
Comment (Fui03) [inline docs/UserGuide.md:40]: I really like how you organized the commands into different categories. It makes the document much easier to navigate and understand!

File: docs/UserGuide.md [hunk 6]
```diff
@@ -36,213 +148,261 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
...
+<a id="how-to-use-track-mate-commands"></a>
 
-* Items with `…`​ after them can be used multiple times including zero times.<br>
-  e.g. `[t/TAG]…​` can be used as ` ` (i.e. 0 times), `t/friend`, `t/friend t/family` etc.
+TrackMate operates primarily through text-based commands. Before we explore the specific commands in detail in the 
+Commands section, let’s familiarize ourselves with the basic components and format of a command.
 
-* Parameters can be in any order.<br>
-  e.g. if the command specifies `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.
+#### 3.3.1 Parameter Prefixes:
```
Comment (Fui03) [inline docs/UserGuide.md:170]: I like how you've presented the parameter prefix! It makes it easy to reference them!

**Review by Fui03 — COMMENTED** [review]
This is a good template for user guide!

File: docs/UserGuide.md [hunk 7]
```diff
@@ -4,16 +4,128 @@ title: "User Guide"
...
+information on how to interpret icons, formatting, and instructions provided throughout the document.
+
+### 2.1 Navigating the Document:
+
+TO BE DONE
+
+### 2.2 Sections:
+<a id="sections"></a>
+
+Below is a detailed overview of the main sections within this User Guide and what can be expected from each section.
```
Comment (tansanchian) [inline docs/UserGuide.md:112]: I like how you mark parts that yet to be done to 'TO BE DONE' so it is easier to be referenced to in the future.

**Review by tansanchian — APPROVED** [review]
LGTM!

**Pull request merged**